src/main/java/com/budgetapp/entities/Expense.java

```diff
@@ -12,6 +13,12 @@ public class Expense extends BaseEntity {
     private String name;
     private BigDecimal value;
 
+    @ManyToOne
```

Comment:
It is good practice to define explicitly what the join column should be as well as proving a unique foreign key name

```suggestion
    @ManyToOne
    @JoinColumn(
            name = "user_id",
            foreignKey = @ForeignKey(name = "users_fk")
    )
```